kratos/geometries/nurbs_curve_geometry.h

```diff
@@ -440,6 +438,31 @@ class NurbsCurveGeometry : public Geometry<typename TContainerPointType::value_t
     ///@name Private Operations
     ///@{
 
+
+    /*
+    * @brief Checks if the knot vector is coinciding with the number of
+    *        control points and the polynomial degree. If the knot vector
+    *        has a multiplicity of p+1 in the beginning, it is reduced to p.
+    */
+    void CheckAndFitKnotVectors()
+    {
+        SizeType number_of_cps = this->size();
+
+        if (mKnots.size() != NurbsUtilities::GetNumberOfKnots(mPolynomialDegree, number_of_cps)) {
+            if (mKnots.size() == NurbsUtilities::GetNumberOfKnots(mPolynomialDegree, number_of_cps - 2)) {
+                Vector Knots = ZeroVector(mKnots.size() - 2);
+                for (SizeType i = 0; i < mKnots.size() - 2; ++i)
```

Comment:
please also add the brackets here
otherwise it is a nasty bug if someone adds another line to this loop